docconv_test.go

```diff
@@ -18,3 +19,77 @@ func TestConvertTrimsSpace(t *testing.T) {
 		t.Errorf("body = %v, want %v", resp.Body, want)
 	}
 }
+
+func TestXMLMaxWord(t *testing.T) {
+	t.Run("max word not set", func(t *testing.T) {
+		checkMaxWord := checkXMLMaxWord()
+		if checkMaxWord != false {
+			t.Fatalf("got %v, want false", checkMaxWord)
+		}
+	})
+	t.Run("test checkMaxWord", func(t *testing.T) {
+		SetConfig(Config{Limitation: LenthLimitation{XMLMaxWord: 10}})
+		checkMaxWord := checkXMLMaxWord()
+		if checkMaxWord != true {
+			t.Fatalf("got %v, want true", checkMaxWord)
+		}
+	})
+	t.Run("test xmlMaxWordExceed", func(t *testing.T) {
+		SetConfig(Config{Limitation: LenthLimitation{XMLMaxWord: 10}})
+		exceed := xmlMaxWordExceed(10)
+		if exceed != false {
+			t.Fatalf("got %v, want false", exceed)
+		}
+		exceed = xmlMaxWordExceed(11)
+		if exceed != true {
+			t.Fatalf("got %v, want true", exceed)
+		}
+	})
+	t.Run("test parse pptx with maxword", func(t *testing.T) {
+		SetConfig(Config{Limitation: LenthLimitation{XMLMaxWord: 2}})
+		f, err := os.Open("./docx_test/testdata/sample_3.docx")
+		if err != nil {
+			t.Fatalf("got error = %v, want nil", err)
+		}
+
+		resp, _, err := ConvertDocx(f)
+		if err != nil {
+			t.Fatalf("got error = %v, want nil", err)
+		}
+		if want := "Content from docx file"; !strings.Contains(resp, want) {
+			t.Errorf("expected %v to contains %v", resp, want)
+		}
+		if want := "second"; strings.Contains(resp, want) {
+			t.Errorf("expected %v to not contains %v", resp, want)
+		}
+	})
+
+}
+
+func TestPDFPageLimit(t *testing.T) {
+	SetConfig(Config{Limitation: LenthLimitation{PdfFirstPage: 2, PdfLastPage: 3}})
+	f, err := os.Open("./pdf_test/testdata/pdf.pdf")
+	if err != nil {
+		t.Fatalf("got error = %v, want nil", err)
+	}
+
+	resp, _, err := ConvertPDF(f)
```

Comment:
@xiaoxin01 I think the best approach for supporting this kind configuration is to introduce new `Converter`s. Something like:

```go
package docconv

type PDFConverter struct {
  pageRange []int
  maxWords int
}

func NewPDFConverter(pageRange []int, maxWords int) *PDFConverter {
  return &PDFConverter{
    pageRange: pageRange,
    maxWords: maxWords,
  }
}

func (c *PDFConverter) Convert(r io.Reader) (string, map[string]string, error) {
  // ...
}
```

We want to avoid anymore global package state so I think a solution like this is probably going to be better. What do you think?